kernel/siw_qp.c

```diff
@@ -638,7 +659,7 @@ siw_qp_modify(struct siw_qp *qp, struct siw_qp_attrs *attrs,
 		break;
 	}
 	if (drop_conn)
-		siw_qp_cm_drop(qp, 0);
+		siw_qp_cm_drop(qp, 1);
```

Comment:
siw_cm_work_handler() calls siw_cep_set_inuse() and then calls siw_proc_mpareply(). If the ird negotiation fails, the state is set to SIW_QP_STATE_TERMINATE which results in drop_conn being set. This causes siw_qp_cm_drop() to be called and it calls siw_cep_set_inuse() which results in a deadlock.

Is there a better way to handle this?